Deployment/asdk-prechecker.ps1

```diff
@@ -0,0 +1,384 @@
+    $selectedDisks = $physicalDisks | Group-Object -Property BusType | Sort-Object -Property Count -Descending | Select-Object -First 1
+
+    if ($selectedDisks.Count -ge 3) {
+        write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Listing of all physical disks meeting ASDK requirements:"
+        write-host -ForegroundColor gray ($physicalDisks | Format-Table -Property @("FriendlyName", "SerialNumber", "BusType", "OperationalStatus", "HealthStatus", "Usage", "Size") | Out-String)
+        write-host -ForegroundColor green "["(date -format "HH:mm:ss")"]" " -- Check physical disks passed successfully. Note that ASDK handles situations where there is a pre-existing storage pool, and will delete/recreate it."
+        $Global:ChecksSuccess++
+    }
+
+    if ($selectedDisks.Count -lt 3) {
+        write-host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- Check physical disks failed - At least 4 disks or more of the same bus type (RAID/SAS/SATA), and of capacity 135 GB or higher are strongly recommended. 3-disk configurations may work but are not tested by Microsoft."
+        $Global:ChecksFailure++
+     }    
+
+}
+
+function CheckFreeSpaceForExtraction {
+
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " Checking free space for extracting the ASDK files..."
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Listing disks and their free space"
+    write-host -ForegroundColor gray (Get-Disk | Get-Partition | Get-Volume | Sort-Object -Property SizeRemaining -Descending | Out-String)
+    $volumes = (Get-disk | ? {$_.BusType -ne 'File Backed Virtual' -or $_.IsBoot} | Get-Partition | Get-Volume |`
+         ? {-not [String]::IsNullOrEmpty($_.DriveLetter)} | sort -Property SizeRemaining -Descending)
+    if (!$volumes -or ($volumes | Measure-Object).count -le 0) {
+        Write-Host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- Free space check failed. No volumes are available."
+        $Global:ChecksFailure++
+    }
+    if ($volumes[0].SizeRemaining -lt 120 * 1024 * 1024 * 1024) {
+        write-host -ForegroundColor red "["(date -format "HH:mm:ss")"]" " -- Free space check failed. ASDK requires 130 GB for the expanded Cloudbuilder.vhdx file. An additional 40 GB may be needed if you want to keep the ZIP and self extractor files."
+        $Global:ChecksFailure++
+    }
+    else
+    {
+        write-host -ForegroundColor green "["(date -format "HH:mm:ss")"]" " -- Free space check passed successfully."
+        $Global:ChecksSuccess++
+    }
+}
+
+function CheckRam {
+    write-host -ForegroundColor yellow "["(date -format "HH:mm:ss")"]" "Checking Memory..."
+    
+    $mem = Get-WmiObject -Class Win32_ComputerSystem
+    $totalMemoryInGB = [Math]::Round($mem.TotalPhysicalMemory / (1024 * 1024 * 1024))
+    write-host -ForegroundColor gray "["(date -format "HH:mm:ss")"]" " -- Memory on this server = $totalMemoryInGB"
+    if ($totalMemoryInGB -lt 96) {
```

Comment:
Currently the memory for ASDK has 192-GB minimum and 256-GB recommended.
Please update the value accordingly.

Reply:
Changed test condition to `$totalMemoryInGB -lt 192`.  Improved error message description.